src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlDecoder.cs

```diff
@@ -0,0 +1,990 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+using System.Text.Unicode;
+using static System.Buffers.Text.Base64;
+
+namespace System.Buffers.Text
+{
+    // AVX2 and Vector128 version based on https://github.com/gfoidl/Base64/blob/master/source/gfoidl.Base64/Internal/Encodings/Base64UrlEncoding.cs
+
+    public static partial class Base64Url
+    {
+        /// <summary>
+        /// Returns the maximum length (in bytes) of the result if you were to decode base 64 encoded text within a byte span of size "base64Length".
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="base64Length"/> is less than 0.
+        /// </exception>
+        public static int GetMaxDecodedLength(int base64Length)
+        {
+            if (base64Length < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
+            }
+
+            int remainder = base64Length % 4;
```

Comment:
```suggestion
            int remainder = (int)((uint)base64Length % 4);
```
To let the JIT emit the bit-hack solution for us.